File: Implementations/DSALG/quick_sort.py
def sort(left, right, pivot):
    return left + [pivot] + right

def quickSort(arr):
    if len(arr) <= 1:
        return arr

    r = []
    l = []
    pivot = arr.pop()
    for i in arr:
        if i > pivot:
            r += [i]
        else:
            l += [i]
    return sort(quickSort(l), quickSort(r), pivot)


arr = [9, 6, 2, 9, 1]

print(quickSort(arr))

File: js_basics/quick_sort.js
function sort(left, right, pivot){
    var arr = left;
    arr.push(pivot);
    arr = arr.concat(right);

    return arr;
}

function quickSort(arr){
    if(arr.length <= 1){
        return arr;
    }

    var r = [];
    var l = [];

    var pivot = arr.shift();
    for(let i = 0; i<arr.length; i++){
        if(arr[i] > pivot){
            r.push(arr[i]);
        }else{
            l.push(arr[i]);
        }
    }

    return sort(quickSort(l), quickSort(r), pivot);
}


var arr = [9, 8, 1,3, 5, 10];

arr = quickSort(arr);

console.log(arr);